### PR TITLE
Add docs

### PR DESCRIPTION
**Pull request by atirut-w**
The docs can be deployed to GH Pages by running `mkdocs gh-deploy`. A workflow to automatically regen and deploy docs can be added, but I have not done it yet on this PR (feel free to ask tho).

### CODE REVIEW

**Comment by atirut-w** [comment]
Ah, right. What do I do with the old docs that are bunched up in README? Do I try to slim it down?

**Comment by atirut-w** [comment]
Now that I think about it, let's not merge this yet. I'd like to actually document the system calls, too.

**Comment by Zeal8bit** [comment]
@atirut-w Thanks for your help on that! Feel free to poke me for any further detail about the implementation or when this draft becomes ready to review

For the syscalls, the `README` explains them from the user application perspective, not driver/kernel perspective, if you think it would make sense to have another sections/chapter/part for that part, or even a subsection in `driver` part, feel free to add it

**Comment by atirut-w** [comment]
> For the syscalls, the `README` explains them from the user application perspective, not driver/kernel perspective, if you think it would make sense to have another sections/chapter/part for that part, or even a subsection in `driver` part, feel free to add it

It's just that the readme only tell you the name, but doesn't go into enough details for some of the more specific syscalls like `mount`, directory operations, time format, etc. Now that there's a dedicated place for docs, I feel like it would be a disservice not to elaborate on them.

**Comment by atirut-w** [comment]
@Zeal8bit I'd like a review of the convention I used to document the system calls with (`docs/system-calls.md`). If they're okay, I can go ahead and do the rest.

**Comment by atirut-w** [comment]
New status: proofreading and feedback pending :)

File: docs/system-calls.md [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+### Parameters
+- `H` - The descriptor to read from
+- `DE` - The buffer to read into
+- `BC` - The number of bytes to read
+
+### Returns
+- `A` - 0 if successful, otherwise an error code
+- `BC` - The number of bytes read
+
+## `write`
```
Comment (Zeal8bit) [inline docs/system-calls.md:44]: Is there a way to separate the syscalls in a nice way? Can we integrate raw HTML code?
Does `mkdocs` allow us to make custom class for a markdown block?

**Comment by Zeal8bit** [comment]
@atirut-w Thanks a lot for your contribution! I love the design of the doc, I think you can integrate an automated task to deploy the pages.

I only left a single comment for the separating the syscalls but nothing very important

File: docs/system-calls.md (hunk 1 above)
Comment (atirut-w) [reply docs/system-calls.md:44]: There's a lot of customization you can do. IIRC there are detail HTML tags and horizontal bars in markdown, and Material for MkDocs provide a lot of [their own stuff](https://squidfunk.github.io/mkdocs-material/reference/). Not quite sure which one to use, though.

**Comment by atirut-w** [comment]
> I think you can integrate an automated task to deploy the pages.

Indeed. I even made my own GH Action for it, but I'm not sure if it would be scope creep for this PR or if you would want to host it somewhere else other than GH Pages.

**Comment by Zeal8bit** [comment]
Thanks for your contribution, merging it!

**Pull request merged**